server/engine/src/main/java/org/eclipse/lsp/cobol/core/visitor/CobolVisitor.java

```diff
@@ -1870,26 +1867,23 @@ private TextExtractionState(ExtendedDocument extendedDocument) {
 
       void update(ProcedureSectionNode section, Token firstToken) {
       if (lastParagraphNode != null) {
-        String text = input.getText(new Interval(firstParagraphToken.getStartIndex(), lastSentenseToken.getStopIndex()));
-        lastParagraphNode.setText(text);
         lastParagraphNode.getLocality().getRange().setEnd(lastSentensePosition);
+        lastParagraphNode.setText(extendedDocument.getBaseText(lastParagraphNode.getLocality()));
```

Comment:
would it be possible to delay this action to the point where it is actually requrested?
And perhaps consider the needs of the only user of this data `CFASTBuilderImpl.cutSnippet`?